### PR TITLE
feat(cannon-native): The great MIPS64 migration

### PR DESCRIPTION
**Pull request by clabby**
## Overview

First man discovered fire, then why 64-bit CPU architectures are better than everything else.

This PR fully ports the native implementation of `cannon` to MIPS64 - all tests beyond EVM differential tests are passing in the `cannon` package.

### Stack

- [x] Native MIPS64 conversion
- [x] Contract MIPS64 conversion
- [x] `op-program` MIPS64 tests

### Merging

Merge #11404 into this branch first after reviews, then merge into `develop`.

### CODE REVIEW

**Comment by clabby** [comment]
* **#11404** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/11404?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#11388** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/11388?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cannon/mipsevm/multithreaded/instrumented_test.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func TestInstrumentedState_MultithreadedProgram(t *testing.T) {
 
 	var stdOutBuf, stdErrBuf bytes.Buffer
 	us := NewInstrumentedState(state, oracle, io.MultiWriter(&stdOutBuf, os.Stdout), io.MultiWriter(&stdErrBuf, os.Stderr), testutil.CreateLogger())
-	for i := 0; i < 1_000_000; i++ {
+	for i := 0; i < 2_000_000; i++ {
```
Comment (Inphi) [inline cannon/mipsevm/multithreaded/instrumented_test.go:40]: Does this mean running this test on 64-bit requires more steps?

Comment (clabby) [reply cannon/mipsevm/multithreaded/instrumented_test.go:40]: Yes!

**Comment by tynes** [comment]
Closing this due to https://github.com/ethereum-optimism/optimism/pull/12029 being merged

**Pull request closed without merging**